habitat/core/env.py

```diff
@@ -213,6 +213,11 @@ def _update_step_stats(self) -> None:
         if self._past_limit():
             self._episode_over = True
 
+        if self.episode_iterator is not None and isinstance(
```

Comment:
While `self.episode_iterator` mentioned as Optional I don't see Env functioning without it here. Maybe check for subclass of `EpisodeIterator`.

Reply:
A separate PR probably makes sense for changing that functionality (dataset is also marked as optional).